Source/Base/ASDisplayNode+Ancestry.h

```diff
@@ -22,7 +22,36 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ASDisplayNode (Ancestry)
 
-- (NSEnumerator *)ancestorEnumeratorWithSelf:(BOOL)includeSelf;
+/**
+ * Returns an object to enumerate the supernode ancestry of this node, starting with its supernode.
+ *
+ * For instance, you could write:
+ *   for (ASDisplayNode *node in self.supernodes) {
+ *     if ([node.backgroundColor isEqual:[UIColor blueColor]]) {
+ *       node.hidden = YES;
+ *     }
+ *   }
+ *
+ * Note: If this property is read on the main thread, the enumeration will attempt to go up
```

Comment:
Isn't the above example code dangerous then? It's casting to an `ASDisplayNode` when it could be either a node or a layer? I don't think the above would crash but it's a little misleading.